libsolidity/lsp/DocumentHoverHandler.cpp

```diff
@@ -0,0 +1,127 @@
+/*
+	This file is part of solidity.
+
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+// SPDX-License-Identifier: GPL-3.0
+
+#include <libsolidity/lsp/DocumentHoverHandler.h>
+#include <libsolidity/lsp/Utils.h>
+
+#include <fmt/format.h>
+
+namespace solidity::lsp
+{
+
+using namespace std;
+
+using namespace solidity::lsp;
+using namespace solidity::langutil;
+using namespace solidity::frontend;
+
+namespace
+{
+
+struct MarkdownBuilder
```

Comment:
My design pattern OCD is acting up - why isn't this a class with a private `result` field, returned by a public `build()` method? Also, I don't see the actual `result` `stringstream` being used anywhere, so the `build()` method could just return a `string`.

Reply:
Of course its used, currently in `110` (`auto tooltipText = markdown.result.str();`). It's not a class but simply a struct to not over-design something simple.

Reply:
Yeah, but I mean the `stringstream` itself is not used directly, rather you're fetching the string from it via `.str()` call.